library/std/src/sys/sync/rwlock/queue.rs

```diff
@@ -418,21 +471,21 @@ impl RwLock {
         // The state was observed with acquire ordering above, so the current
         // thread will observe all node initializations.
 
-        // SAFETY:
-        // Because new read-locks cannot be acquired while threads are queued,
-        // all queue-lock owners will observe the set `LOCKED` bit. Because they
-        // do not modify the queue while there is a lock owner, the queue will
-        // not be removed from here.
+        // FIXME this is a bit confusing
+        // SAFETY: Because new read-locks cannot be acquired while threads are queued, all
+        // queue-lock owners will observe the set `LOCKED` bit. And because no downgrade can be in
+        // progress (we checked above), they hence do not modify the queue, so the queue will not be
+        // removed from here.
```

Comment:
> Because new read-locks cannot be acquired while threads are queued, all queue-lock owners will observe the set `LOCKED` bit.

I think this is a tiny bit confusing to me, as it's not clear why read-locking would cause lock owners to not observe the `LOCKED` bit, and also  it's not clear why not observing the set `LOCKED` bit would lead to threads modifying the queue. Do you think you could clarify this?